### PR TITLE
Don't try to update viewer state if it was deleted

### LINKED ISSUE #2521: Plot Option breaks when viewer deleted while selected
**Reporter:** pllim

Workflow to reproduce:

1. Fire up `ImvizExample.ipynb`
2. Load both images. Do not have to relink them (not relevant here).
3. Open Plot Options.
4. Create a new viewer.
5. For clarity, while `imviz-0` is still selected in Plot Option, select image A. Then load image B into the new viewer (`imviz-1`).
6. Select `imviz-1` in Plot Options.
7. Delete the viewer you just created.

What actually happened: Now the viewer dropdown is gone from Plot Options, but the data dropdown is stuck in B. There is no way to switch back to A (maybe because the dropdown options is stuck at `imviz-1` that only had B loaded). See attached screenshot.

Screenshot 2023-10-18 222206

What should happen: Data dropdown should automatically switch back to imviz-0 and have both A and B available.

[🐱](https://jira.stsci.edu/browse/JDAT-3899)

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter.

### PR DESCRIPTION
**Pull request by rosteen**
Closes #2521 , turned out to be very simple once I tracked it down. This was previously causing an error if the viewer was deleted, since `None` has no `state`.

### CODE REVIEW

**Comment by pllim** [comment]
Add a test? 😉

File: jdaviz/configs/default/plugins/plot_options/plot_options.py [hunk 1]
```diff
@@ -660,9 +660,11 @@ def _update_stretch_histogram(self, msg={}):
             if isinstance(viewer_label_old, list):
                 viewer_label_old = viewer_label_old[0]
             if len(viewer_label_old):
-                vs_old = self.app.get_viewer(viewer_label_old).state
-                for attr in ('x_min', 'x_max', 'y_min', 'y_max'):
-                    vs_old.remove_callback(attr, self._update_stretch_histogram)
+                # If the previously selected viewer was deleted, we don't need to do this.
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/plot_options/plot_options.py:663]: Can these two lines just be combined into `if viewer_label_old in self.app._viewer_store` and avoid a whole level of indentation?  If it doesn't have length (so an empty string), then it should fail that check as well, right?

**Comment by rosteen** [comment]
> Add a test? 😉

No.

...ok, fiiine 😆

**Review by kecnry — APPROVED** [review]

**Comment by rosteen** [comment]
I'm having trouble writing a test that fails on `main`. The code from the test I added here results in an error traceback on `main` if run in a notebook context, but passes as a test. Not sure what's missing from the test to get it to fail...

**Comment by rosteen** [comment]
As discussed offline, going to merge this rather than spending a bunch more time figuring out a test that fails on `main`. Codecov also seems to be giving a wonky result.

**Pull request merged**